pilot/pkg/networking/core/v1alpha3/cluster_cache.go

```diff
@@ -124,6 +126,8 @@ func (t *clusterCache) Key() any {
 	}
 	h.Write(Separator)
 
+	t.endpointBuilder.WriteHash(h)
```

Comment:
why do we need to add more factor for cache key calculation when we  just do a cleanup

Reply:
My concern is mostly that `EndpointBuilder` takes a `model.Proxy` and `PushContext`. It can decide to use whatever parts of that it needs. The cache key for EndpointBuilder will be updated to include those, but now that affects CDS. 
I want to make it hard(er) to write caching bugs. 

Right now endpoint builder has these things that CDS dosn't:

* isClusterLocal
* node type
* failover priority labels

> we just do a cleanup

I'm not sure what you mean.

Reply:
My concern is mostly that `EndpointBuilder` takes a `model.Proxy` and `PushContext`. It can decide to use whatever parts of that it needs. The cache key for EndpointBuilder will be updated to include those, but now that affects CDS. 
I want to make it hard(er) to write caching bugs. 

Right now endpoint builder has these things that CDS dosn't:

* isClusterLocal
* node type
* failover priority labels

> we just do a cleanup

I'm not sure what you mean.